### PR TITLE
Only remove update listener when unload was succesfull

### PR DESCRIPTION
**Pull request by iMicknl**

### CODE REVIEW

File: custom_components/tahoma/__init__.py [hunk 1]
```diff
@@ -181,6 +179,7 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
     )
 
     if unload_ok:
+        hass.data[DOMAIN][entry.entry_id]["update_listener"]()
```
Comment (tetienne) [inline custom_components/tahoma/__init__.py:182]: What's the point already to call this method?

Comment (iMicknl) [reply custom_components/tahoma/__init__.py:182]: We register an 'update listener' with our callback function, to get notified on changes. We need to unregister this callback when the component is unloaded.

Comment (tetienne) [reply custom_components/tahoma/__init__.py:182]: I mean, why calling it, will unregister it?

Comment (iMicknl) [reply custom_components/tahoma/__init__.py:182]: Yes! Sorry, I misinterpreted your question. Have a look at the following function definition.

```
    def add_update_listener(self, listener: UpdateListenerType) -> CALLBACK_TYPE:
        """Listen for when entry is updated.

        Returns function to unlisten.
        """
```

**Review by tetienne — APPROVED** [review]

**Pull request merged**